packages/build/tests/plugins/tests.js

```diff
@@ -86,6 +86,12 @@ test('Resolution is relative to the build directory', async (t) => {
   t.snapshot(normalizeOutput(output))
 })
 
+test('Resolution respects monorepo node module resolution rules', async (t) => {
+  const fixture = await new Fixture('./fixtures/monorepo')
+  const output = await fixture.withFlags({ packagePath: 'apps/unpinned' }).runWithBuild()
+  t.assert(output.indexOf('@8.5.3') > 0)
```

Comment:
Maybe to just add a bit more information on the setup/assertion
```suggestion
  // fixture has 2 versions of the same build plugin used by different workspaces
  // this ensures version used by apps/unpinned is used instead of version that
  // is hoisted in shared monorepo node_modules
  t.assert(output.indexOf('@8.5.3') > 0)
```